homeassistant/components/weenect/entity.py

```diff
@@ -0,0 +1,49 @@
+"""The Weenect base class."""
+from __future__ import annotations
+
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+)
+
+from .const import ATTRIBUTION, DOMAIN, TRACKER_REMOVED
+
+
+class WeenectBaseEntity(CoordinatorEntity):
+    """Abstract base entity for weenect."""
+
+    def __init__(self, coordinator: DataUpdateCoordinator, tracker_id: int) -> None:
+        """Init Base Entity for Weenect entities."""
+        super().__init__(coordinator)
+        self.id = tracker_id
+        self._attr_attribution = ATTRIBUTION
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, str(self.id))},
+            name=str(self.coordinator.data[self.id]["name"]),
+            model=str(self.coordinator.data[self.id]["type"]),
+            manufacturer="Weenect",
+            sw_version=str(self.coordinator.data[self.id]["firmware"]),
+        )
+        self._attr_extra_state_attributes = {
+            "id": self.id,
+            "sim": str(self.coordinator.data[self.id]["sim"]),
+            "imei": str(self.coordinator.data[self.id]["imei"]),
+        }
```

Comment:
`self.coordinator.data[self.id]` could maybe be lifted into a local in this function.